src/daft-csv/src/read.rs

```diff
@@ -168,29 +243,95 @@ async fn read_csv_single_into_table(
             .unwrap()
     });
     // Collect all chunks in chunk x column form.
-    let chunks = chunk_stream
+    let tables = chunk_stream
         // Limit the number of chunks we have in flight at any given time.
-        .try_buffered(max_chunks_in_flight)
+        .try_buffered(max_chunks_in_flight);
+
+    // Fields expected for read and evaluating predicate
+    let read_fields = if let Some(read_columns) = &read_columns {
+        let field_map = fields
+            .iter()
+            .map(|field| (field.name.clone(), field.clone()))
+            .collect::<HashMap<String, Field>>();
```

Comment:
If we're going to need to clone the field anyways when constructing `read_fields` in the next statement, we might as well build the hashmap on references so we don't needless clone each field twice.
```suggestion
            .map(|field| (field.name.as_ref(), field))
            .collect::<HashMap<&str, &Field>>();
```